app/user.go

```diff
@@ -2141,3 +2141,39 @@ func (a *App) invalidateUserCacheAndPublish(userId string) {
 func (a *App) GetKnownUsers(userID string) ([]string, *model.AppError) {
 	return a.Srv().Store.User().GetKnownUsers(userID)
 }
+
+// ConvertBotToUser converts a bot to user.
+func (a *App) ConvertBotToUser(bot *model.Bot, userPatch *model.UserPatch, sysadmin bool) (*model.User, *model.AppError) {
+	user, err := a.Srv().Store.User().Get(bot.UserId)
+	if err != nil {
+		return nil, err
+	}
+
+	if sysadmin && !user.IsInRole(model.SYSTEM_ADMIN_ROLE_ID) {
+		if _, err = a.UpdateUserRoles(
```

Comment:
Probably I'm missing something important but why are we converting a bot into a sysadmin?

nit: In multiline checks for me is more readable to use:

```go
_, err = a.Update...()
if err != nil {}
```

Reply:
This was present in the legacy cli command 🤷‍♂️

Reply:
Yeah but we should know why because giving sysadmin role to a converted bot seems a bit odd and at least we should be sure

Reply:
I think it makes sense to upgrade a bot to system admin user. I don't know the exact reason but it can be real life scenario.